grype/db/v6/store.go

```diff
@@ -0,0 +1,61 @@
+package v6
+
+import (
+	"fmt"
+
+	"gorm.io/gorm"
+
+	"github.com/anchore/grype/grype/db/internal/gormadapter"
+	"github.com/anchore/grype/internal/log"
+)
+
+type store struct {
+	*dbMetadataStore
+	db     *gorm.DB
+	config Config
+	write  bool
+}
+
+func newStore(cfg Config, write bool) (*store, error) {
+	db, err := gormadapter.Open(cfg.DBFilePath(), write)
+	if err != nil {
+		return nil, err
+	}
+
+	db.Exec("PRAGMA foreign_keys = ON")
+
+	if write {
+		if err := migrate(db, models()...); err != nil {
+			return nil, fmt.Errorf("unable to migrate: %w", err)
+		}
+	}
+
+	return &store{
+		dbMetadataStore: newDBMetadataStore(db),
+		db:              db,
+		config:          cfg,
+		write:           write,
+	}, nil
+}
+
+func migrate(db *gorm.DB, models ...any) error {
```

Comment:
take or leave feedback: `migrate` sounds like we're changing an existing database schema, but really we're migrating from an empty database to one with all our models in it. A name with `create` in it might make more sense?

Reply:
it's too small of a function anyway to be useful, I rolled it into the `Open` function